packages/react-component-library/src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useRef } from 'react'
+import classNames from 'classnames'
+import TetherComponent from 'react-tether'
+
+import {
+  POPOVER_CLOSE_DELAY,
+  POPOVER_PLACEMENT,
+  POPOVER_PLACEMENTS,
+} from './constants'
+
+import {
+  FloatingBox,
+  FloatingBoxProps,
+  FLOATING_BOX_SCHEME,
+} from '../../primitives/FloatingBox'
+
+interface PopoverProps
+  extends Omit<FloatingBoxProps, 'onMouseEnter' | 'onMouseLeave'> {
+  children: React.ReactElement
```

Comment:
One thing I was thinking is whether it's more intuitive that the `children` is the content and then there is a prop for the `trigger`. Just a thought..

Reply:
The component tree made more sense to me with Target as child, as it more closely represented the pre-existing component tree had there not been a Popover. I thought of Popover as a wrapper component that enhanced a target (child).

```
Window 
   > Document 
      > Container 
        > Popover (Popover content as prop)
          > Target 

vs

Window 
   > Document 
      > Container 
         > Popover (Target as prop)
            > Content?
```

Popover is an enhancement of Target not the other way around? I could maybe implement it as a HOC instead. And you enhance target components.

Something like:

```js
const TargetComponentWithPopover = withPopover(TargetComponent, {
  // options
})
```

Reply:
I'm going to implement this enhancer as part of another PR in the near future. Marking as resolved for now.